main.go

```diff
@@ -177,17 +178,19 @@ func main() {
 	retryPeriod := time.Second * time.Duration(leaderLeaseRetryPeriod)
 
 	mgr, err := ctrl.NewManager(kubeConfig, ctrl.Options{
-		SyncPeriod:              &syncPeriod,
-		Scheme:                  scheme,
-		MetricsBindAddress:      metricsAddr,
-		Port:                    9443,
-		LeaderElection:          enableLeaderElection,
-		LeaseDuration:           &leaseDuration,
-		RenewDeadline:           &renewDeadline,
-		RetryPeriod:             &retryPeriod,
-		LeaderElectionID:        config.LeaderElectionKey,
-		LeaderElectionNamespace: config.LeaderElectionNamespace,
-		HealthProbeBindAddress:  ":61779", // the liveness endpoint is default to "/healthz"
+		SyncPeriod:                 &syncPeriod,
+		Scheme:                     scheme,
+		MetricsBindAddress:         metricsAddr,
+		Port:                       9443,
+		LeaderElection:             enableLeaderElection,
+		LeaseDuration:              &leaseDuration,
+		RenewDeadline:              &renewDeadline,
+		RetryPeriod:                &retryPeriod,
+		LeaderElectionID:           config.LeaderElectionKey,
+		LeaderElectionNamespace:    config.LeaderElectionNamespace,
+		LeaderElectionResourceLock: resourcelock.ConfigMapsResourceLock,
```

Comment:
What's the advantage/downside of using the default leader election locking?

Reply:
< controller-runtime v0.7 had the value set to "configmaps", so was setting to the same. But verified that the new default "configmapsleases" also works.
Will update. 
Ref: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L156